### PR TITLE
Do not mention anymore file templates

### PR DESCRIPTION
**Pull request by dadoonet**
From this [discussion](https://discuss.elastic.co/t/not-possible-yet-another-how-to-pre-register-a-pretty-printed-search-template-with-conditional-clauses/189144), we should not mention anymore using file templates.

### CODE REVIEW

File: docs/reference/search/search-template.asciidoc [hunk 1]
```diff
@@ -343,7 +343,7 @@ We could write the query as:
 ==================================
 As written above, this template is not valid JSON because it includes the
 _section_ markers like `{{#line_no}}`.  For this reason, the template should
-either be stored in a file (see <<pre-registered-templates>>) or, when used
+either be stored in elasticsearch (see <<pre-registered-templates>>) or, when used
```
Comment (jrodewig) [inline docs/reference/search/search-template.asciidoc]:
```suggestion
either be stored in {es} (see <<pre-registered-templates>>) or, when used
```

`{es}` will expand to `Elasticsearch.` If you prefer not to use an attribute, please capitalize `Elasticsearch`.

**Review by jrodewig — APPROVED** [review]
Looks good. Made one minor correction.

**Comment by hyksos** [comment]
I don't think this makes sense.
You can't store such a template (not valid JSON) in Elasticsearch UNLESS you write it as a string and escape everything. Because the only way to load template into ES is via the API now and of course you can't push invalid JSON into the API. There is only one option now for people who have templates that are not JSON because they use mustache functions outside of json strings.

The alternative, which was that if it wasn't valid JSON you could put it in a file (as invalid JSON) and reference it is now totally gone.

This is evidenced by the fact that if you follow the link for "Pre-registered template" it goes to the section that explains how to push templates with the _scripts API... You can't push invalid JSON through that of course.
Also the text at the top of "Conditional clauses" says:
"Conditional clauses cannot be expressed using the JSON form of the template. Instead, the template must be passed as a string."
That says the same thing as me... it MUST be passed as a string(escaped) because it's not valid JSON.

**Comment by dadoonet** [comment]
I see. That's true. I didn't read the full paragraph apparently but was just trying to remove the reference to the file based templates.
We should probably come with something else. May be remove this part of the text.

**Comment by jrodewig** [comment]
@dadoonet @hyksos I pushed 9de856e to rewrite parts of the **Conditional clauses** section.

Please let me know your thoughts. Thanks!

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
Closing as this PR has stalled and gotten stale. I've opened #50089 to track the issue.

**Pull request closed without merging**